load_installed_packages: return full dependency tree

zprogd opened the pull request:
Format:
```
my $installed_packages =
{
    "perl/Scalar-List-Utils" => {
        package => bless( {
                category     => "perl",
                is_bootstrap => 0,
                name         => "Scalar-List-Utils",
                release      => 1,
                version      => 1.47,
            }, "Pakket::PackageQuery" ),
        prereqs => {
            "perl/Test-Simple" => bless({
                    category  => "perl",
                    name      => "Test-Simple",
                    version   => 0
                }, "Pakket::Requirement"),
        },
        used_by =>{
                "perl/File-Temp"   => 1,
                "perl/Task-Weaken" => 1,
                "perl/Test-Simple" => 1
            },
    },
};
```

xsawyerx commented:
I need you to provide more details on this. This mixes cleanup (`force_reinstall` -> `force`), changing the Requirement object, and introducing a structure with both Requirement and PackageQuery objects.

You need to split the commits and provide an explanation for this.

zprogd commented:
@xsawyerx 
1. moved 'rename force_reinstall -> force' to another pull request https://github.com/xsawyerx/pakket/pull/165
2. I need short_name()  in Pakket::Requirement. I moved that function inside Requirement
3. Dependency tree of installed packages should describe:
- each installed package(category/name/version/revision), I'm using Pakket::PackageQuery
- prerequisites for each package(category/name/versionrange), I'm using Pakket::Requirement
- list of packages which using package, I'm using list of short_names.
- all those lists are indexed by short_name for quick access

zprogd commented:
This commit https://github.com/xsawyerx/pakket/commit/4918631224c1cd001688256351bf8705bb17a5f6 will use that list of installed packages
